edx_rest_framework_extensions/middleware.py

```diff
@@ -31,19 +31,29 @@ def _add_missing_jwt_permission_classes(self, view_class):
         Adds permissions classes that should exist for Jwt based authentication,
         if needed.
         """
-        view_permission_classes = getattr(view_class, 'permission_classes', tuple())
+        view_permissions = list(getattr(view_class, 'permission_classes', []))
 
-        for perm_class in self._required_permission_classes:
+        # Not all permissions are classes, some will be ConditionalPermission
+        # objects from the rest_condition library. So we have to crawl all those
+        # and expand them to see if our target classes are inside the
+        # conditionals somewhere.
+        permission_classes = []
+        while view_permissions:
+            permission = view_permissions.pop()
+            if not hasattr(permission, 'perms_or_conds'):
+                permission_classes.append(permission)
+            else:
+                for child in getattr(permission, 'perms_or_conds', []):
+                    view_permissions.append(child)
 
-            if not self._includes_base_class(view_permission_classes, perm_class):
+        for perm_class in self._required_permission_classes:
+            if not self._includes_base_class(permission_classes, perm_class):
                 log.warning(
-                    u"The view %s allows Jwt Authentication but needs to include the %s permission class.",
+                    u"The view %s allows Jwt Authentication but needs to include the %s permission class (adding it for you)",
                     view_class.__name__,
                     perm_class.__name__,
                 )
-
-        view_class.permission_classes = view_permission_classes
-        view_class.permission_classes += self._required_permission_classes
+                view_class.permission_classes += self._required_permission_classes
```

Comment:
Self note: we should append the missing classes one at a time, not blindly append the list.

Reply:
Is this so we prevent duplicates?  And also in case the permission existed in ConditionalPermission?

Reply:
Yes.

Reply:
Expanded version for others who might be watching: moving it into the inner loop like this means that we're no longer always appending it at the end. The assumption being that if you mention `JwtScope` somewhere, even nested in a conditional, then you've taken it into account and you know what you're doing, so we're not going to append it at the end since it will almost surely ruin any logic you meant to put in there.

But when I first moved it, I left the code the same, which means I would be appending the whole list of required permissions every time a missing permission was found. Which is boolean logic-wise a no-op (ANDing the same condition twice), and we currently only have one required permission -- but it's still redundant and bad practice.